packages/date-time-utilities/src/dateMath/dateMath.ts

```diff
@@ -303,6 +303,22 @@ export function getStartDateOfWeek(date: Date, firstDayOfWeek: DayOfWeek): Date
   return addDays(date, daysOffset);
 }
 
+/**
+ * Gets the date for the last day of the week based on the given date assuming
+ * the specified first day of the week.
+ * @param date - The date to find the beginning of the week date for.
+ * @returns A new date object representing the first day of the week containing the input date.
+ */
+export function getEndDateOfWeek(date: Date, firstDayOfWeek: DayOfWeek): Date {
+  const lastDayOfWeek = firstDayOfWeek - 1 >= 0 ? firstDayOfWeek - 1 : TimeConstants.DaysInOneWeek - 1;
```

Comment:
Curious:
wouldn't modulo operation be more compact?

Reply:
Also applies to calculation below.

Reply:
Yeah, makes sense :)

Reply:
Aha, Javascript modulo would return negative numbers, so we've agreed with @rymeskar to keep it like it is now :)